src/step-input.scss

```diff
@@ -0,0 +1,153 @@
+@import "./settings";
+@import "./new-settings";
+@import "./mixins";
+@import "./functions";
+@import "./button";
+
+$block: #{$fd-namespace}-step-input;
+
+.#{$block} {
+  $fd-step-input-border-thin-width: 0.0625rem;
+  $fd-step-input-border-thick-width: 2 * $fd-step-input-border-thin-width;
+  $fd-input-group-button-active-focus-color: var(--sapContent_ContrastFocusColor);
+  $fd-button-icon-font-size: 0.875rem;
+  $fd-button-icon-font-size-cozy: 1rem;
+
+  @mixin iconSize($size) {
+    &::before,
+    &::after {
+      font-size: $size;
+    }
+  }
+
+  width: auto;
+  display: inline-block;
+  line-height: 0;
+
+  @include fd-fiori-focus() {
+    outline-color: $fd-input-group-button-active-focus-color;
+  }
+
+  @include fd-reset();
+  @include fd-rtl();
+
+  &__content-wrapper {
+    width: auto;
+    min-width: 2.75rem;
+    margin-top: 0.25rem;
+    margin-bottom: 0.25rem;
+    padding: 0;
+    box-sizing: border-box;
+    vertical-align: middle;
+    display: inline-block;
+    position: relative;
+
+    &.is-readonly {
+      outline: none;
+    }
+  }
+
+  &__content-wrapper:focus-within {
+    box-shadow: none;
+    outline-offset: -0.1875rem;
+    outline-width: var(--sapContent_FocusWidth);
+    outline-color: var(--sapContent_FocusColor);
+    outline-style: var(--sapContent_FocusStyle);
+
+    &.is-readonly {
+      outline: none;
+    }
+  }
+
+  &__button-container {
+    width: 2.125rem;
+    min-width: 2rem;
+    vertical-align: middle;
+    display: inline-block;
+    outline: none;
+    height: 100%;
+
+    &.is-readonly {
+      display: none;
+    }
+
+    @include fd-hover() {
+      @include buttonContainerHover();
+    }
+  }
+
+  &__button-container--compact {
+    width: 1.5rem;
+    min-width: 1.5rem;
+  }
+
+  .#{$fd-namespace}-button.#{$block}__button {
+    @include fd-button-reset();
```

Comment:
We can remove ` @include fd-button-reset();` because that's already included in `fd-button`